### PR TITLE
Introduce DT Field and refactor Asset Key Interface

### PR DESCRIPTION
**Pull request by jacques-n**
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/701)

### CODE REVIEW

File: versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java [hunk 1]
```diff
@@ -18,19 +18,24 @@
...
 import com.dremio.nessie.versioned.store.Id;
+import com.dremio.nessie.versioned.store.SaveOp;
 import com.google.common.base.Preconditions;
 
 abstract class PersistentBase<C extends BaseValue<C>> implements HasId {
 
   //unchanging but only generated once needed.
   private Id id;
 
+  private long dt;
```
Comment (snazy) [inline versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java:29]:
```suggestion
  private final long dt;
```

**Review by snazy — CHANGES_REQUESTED** [review]
Left a couple of suggestions via github

**Comment by snazy** [comment]
Hum - seems Reviewable doesn't like additional reviews via github or the other way around. Anyway, left a few comments via Reviewable as well

**Review by rymurr — COMMENTED** [review]
> Reviewed 44 of 44 files at r1.
> *[Reviewable](https://reviewable.io/reviews/projectnessie/nessie/701)* status: all files reviewed, 8 unresolved discussions (waiting on @jacques-n)

___
*[versioned/spi/src/main/java/com/dremio/nessie/versioned/ValueWorker.java, line 20 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MRPh48l0_QHZygL7odC:-MRPh48mAE_qthMnf_Mk:b-qg2z5w) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/spi/src/main/java/com/dremio/nessie/versioned/ValueWorker.java#L20)):*
> ```Java
> import java.util.stream.Stream;
> 
> public interface ValueWorker<VALUE> extends Serializer<VALUE> {
> ```

nit: javadoc
___
*[versioned/tests/src/main/java/com/dremio/nessie/versioned/StringSerializer.java, line 30 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MRPiqf86TqsBwKfi5XP:-MRPiqf86TqsBwKfi5XQ:b7o4vbs) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tests/src/main/java/com/dremio/nessie/versioned/StringSerializer.java#L30)):*
> ```Java
>  * Serializer and ValueHelper implementation for {@code String class}.
>  */
> public final class StringSerializer implements ValueWorker<String> {
> ```

nit: `StringWorker` as its now a worker

**Comment by rymurr** [comment]
A few nits but overall I like the AssetKey change.

**Review by jacques-n — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/projectnessie/nessie/701)* status: 35 of 45 files reviewed, 6 unresolved discussions (waiting on @rymurr and @snazy)

___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/tiered/builder/BaseValue.java, line 39 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MRMc7cNmT1_nfSM8sct-r1-39:-MRRU0q1Dp81Ettfn5fz:b-896fix) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/tiered/builder/BaseValue.java#L39)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

```suggestion
   * Set the date-time in microseconds since epoch when this object was last modified (inserted/updated).
```
</blockquote></details>

Done.
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/tiered/builder/BaseValue.java, line 40 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MRMc7cNmT1_nfSM8sct-r1-40:-MRRU3wT7FXchiUzc-ub:b-896fix) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/tiered/builder/BaseValue.java#L40)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

```suggestion
   * @param dt The date+time in microseconds since epoch
```
</blockquote></details>

Done.
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/DT.java, line 20 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MRMc7cNmT1_nfSM8sdC-r1-20:-MRRUAJWAg26wo2KGJJ0:b-896fix) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/DT.java#L20)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

```suggestion
final class DT {
```
It's a utility class only used in `c.d.n.versioned.impl`, let's make it package-protected `final`
</blockquote></details>

Done.
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/DT.java, line 27 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MRMc7cNmT1_nfSM8sdC-r1-27:-MRRUI001_yPNr0pFK8U:b-896fix) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/DT.java#L27)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

```suggestion
  static long now() {
```
</blockquote></details>

Done.
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/Ids.java, line 20 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MRMc7cNmT1_nfSM8sdD-r1-20:-MRRUL8Z3hPCBCrK9N5m:b-896fix) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/Ids.java#L20)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

```suggestion
public final class Ids {
```
It's a utility class, let's make it `final`

</blockquote></details>

Done.
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java, line 29 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MRMc7cNmT1_nfSM8sd2-r1-29:-MRRUOs50J6YE4r8fT8m:b-896fix) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java#L29)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

```suggestion
  private final long dt;
```
</blockquote></details>

Done.

**Review by rymurr — CHANGES_REQUESTED** [review]
> Reviewed 5 of 10 files at r2.
> *[Reviewable](https://reviewable.io/reviews/projectnessie/nessie/701)* status: 40 of 45 files reviewed, 7 unresolved discussions (waiting on @jacques-n, @rymurr, and @snazy)

___
*[versioned/spi/src/main/java/com/dremio/nessie/versioned/ValueWorker.java, line 33 at r2](https://reviewable.io/reviews/projectnessie/nessie/701#-MRUnon-0dBfcRIffu2-:-MRUnon-0dBfcRIffu20:b-zhso3x) ([raw file](https://github.com/projectnessie/nessie/blob/828a32ac526dd09c07dba8fb8190546976b463c1/versioned/spi/src/main/java/com/dremio/nessie/versioned/ValueWorker.java#L33)):*
> ```
> track multiple types of assets under a single AssetKey heading. 
> ```

I am not sure what you mean here. In my mind an iceberg table would return a stream of `AssetKey` with 1 key per individual file. When you say 'AssetKey heading' do you mean 1 asset key would have a set of related files?

**Comment by rymurr** [comment]
I am happy with this change bar 1 question on the meaning of the javadoc. Think we can close this one out after that.

**Review by snazy — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/projectnessie/nessie/701)* status: 40 of 45 files reviewed, 15 unresolved discussions (waiting on @jacques-n and @rymurr)

___
*[versioned/spi/src/main/java/com/dremio/nessie/versioned/AssetKey.java, line 28 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROwzCP7Ix10XIqwHTG:-MROwzCP7Ix10XIqwHTH:b5lu43y) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/spi/src/main/java/com/dremio/nessie/versioned/AssetKey.java#L28)):*
> ```Java
>  * <p>Assets are pointers in a reference tree. The purpose of an Asset object is to expose the items pointed to by a stored value
>  */
> public abstract class AssetKey {
> ```

Everything in here is `public abstract`, so `interface` might be better than `abstract class`
___
*[versioned/spi/src/main/java/com/dremio/nessie/versioned/AssetKey.java, line 51 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROxAhqAcT35ZL0pIRF:-MROxAhqAcT35ZL0pIRG:b7i7pcg) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/spi/src/main/java/com/dremio/nessie/versioned/AssetKey.java#L51)):*
> ```Java
> 
>   @Override
>   public abstract boolean equals(Object other);
> ```

I assume, `equals` and `hashCode` are intentionally overridden to indicate that those are mandatory for `AssetKey`s.
A hint in the class' Javadoc emphasizing that those must be implemented would be great.
___
*[versioned/tiered/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoBaseValue.java, line 96 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROvmB-1ax-2S4-p-3W:-MROvmB-1ax-2S4-p-3X:b-sq8etb) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoBaseValue.java#L96)):*
> ```Java
>   }
> 
>   void serializeLong(String property, long value) {
> ```

Can you change this to either let all users of `writeInt64` use this method or specialize it to make clear it's only used for "timestamp in microseconds since epoc"?
___
*[versioned/tiered/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoL1.java, line 92 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROvBtSE7pixwMMUf9l:-MROvBtTD1oeQphuPhik:b-td3k93) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoL1.java#L92)):*
> ```Java
>     bsonWriter.writeBoolean(IS_CHECKPOINT, false);
>     serializeId(ORIGIN, checkpointId);
>     bsonWriter.writeInt64(DISTANCE, distanceFromCheckpoint);
> ```

Mind changing this to the `serializeLong()` function you've added?
___
*[versioned/tiered/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoRef.java, line 236 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROvQX14vrPhX4e_-7j:-MROvQX14vrPhX4e_-7k:b-td3k93) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoRef.java#L236)):*
> ```Java
>   private static void serializeDelta(BsonWriter writer, int position, Id oldId, Id newId) {
>     writer.writeStartDocument();
>     writer.writeInt64(POSITION, position);
> ```

Mind changing this to the `serializeLong()` function you've added?
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/tiered/builder/BaseValue.java, line 35 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROo0bJ4K90_HtzEkmQ:-MROo0bJ4K90_HtzEkmR:b-ry4txl) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/tiered/builder/BaseValue.java#L35)):*
> ```Java
>    */
>   T id(Id id);
> ```

Nit: two superfluous empty lines
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/Ids.java, line 20 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROpzKx0WxXDB7Jil06:-MROpzKx0WxXDB7Jil07:b2edoxk) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/Ids.java#L20)):*
> ```Java
> import com.dremio.nessie.versioned.store.Id;
> 
> public class Ids {
> ```

The methods in this class are never used. Missing a refactoring?
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/Ids.java, line 20 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MRP2KnC1bbHN36YXaJk:-MRP2KnC1bbHN36YXaJl:b-bhd4ku) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/Ids.java#L20)):*
> ```Java
> import com.dremio.nessie.versioned.store.Id;
> 
> public class Ids {
> ```

Since `DT`and `Ids` are rather small, internal classes that provide simple utility methods, mind combining those?
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/InternalL1.java, line 197 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROsU5qB1moXHi4-l2q:-MROsU5qB1moXHi4-l2r:b-uejx0n) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/InternalL1.java#L197)):*
> ```Java
>       return this;
>     }
> ```

Unnecessarily added empty line
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/InternalL1.java, line 221 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROytlN8cCFzRPLGN2e:-MROytlN8cCFzRPLGN2f:b8936qo) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/InternalL1.java#L221)):*
> ```Java
> 
>     @Override
>     public Builder dt(long dt) {
> ```

I think, both `id()` and `dt()`should be moved to `PersistentBase`.
I've pushed that here: https://github.com/snazy/nessie/commit/40113cce7325b95cece4fd9c8c064d5f0fc5f047
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java, line 28 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROyiQM5Y6ycsi3T7FU:-MROyiQM5Y6ycsi3T7FV:bfximzv) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java#L28)):*
> ```Java
>   //unchanging but only generated once needed.
>   private Id id;
> ```

Nit: unnecessary empty line
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java, line 61 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROr0Ka6jCZpjo7tqtw:-MROr0Ka6jCZpjo7tqtx:b-a1hqb0) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java#L61)):*
> ```Java
>   }
> 
>   abstract <E extends PersistentBase<C>> EntityType<C, E, ?> getEntityType();
> ```

Implementations of this method are `public`, but should be package-private
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java, line 63 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROrCH62mu-QP15F0M9:-MROrCH7CUuCrrQwcELH:b-cglsdc) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java#L63)):*
> ```Java
>   abstract <E extends PersistentBase<C>> EntityType<C, E, ?> getEntityType();
> 
>   SaveOp<C> toSaveOp() {
> ```

This method isn't used.
___
*[versioned/tests/src/main/java/com/dremio/nessie/versioned/StringSerializer.java, line 60 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROwHJ5AjB0gQEfWaGN:-MROwHJ5AjB0gQEfWaGO:bmyvkrl) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tests/src/main/java/com/dremio/nessie/versioned/StringSerializer.java#L60)):*
> ```Java
>   }
> 
>   @SuppressWarnings("unchecked")
> ```

The suppression is unnecessary

**Comment by snazy** [comment]
Oh man, Reviewable didn't push all my comments the last time. Sorry for the late notice.

**Review by jacques-n — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/projectnessie/nessie/701)* status: 30 of 47 files reviewed, 14 unresolved discussions (waiting on @rymurr and @snazy)

___
*[versioned/spi/src/main/java/com/dremio/nessie/versioned/AssetKey.java, line 28 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROwzCP7Ix10XIqwHTG:-MRVl4EV9M4iv5Ymlw_7:b-1u3eo1) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/spi/src/main/java/com/dremio/nessie/versioned/AssetKey.java#L28)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

Everything in here is `public abstract`, so `interface` might be better than `abstract class`
</blockquote></details>

This is so we can force overriding of hashcode and equals.
___
*[versioned/spi/src/main/java/com/dremio/nessie/versioned/AssetKey.java, line 51 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROxAhqAcT35ZL0pIRF:-MRWo0aT6JR0fKhEUfWY:b-896fix) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/spi/src/main/java/com/dremio/nessie/versioned/AssetKey.java#L51)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

I assume, `equals` and `hashCode` are intentionally overridden to indicate that those are mandatory for `AssetKey`s.
A hint in the class' Javadoc emphasizing that those must be implemented would be great.
</blockquote></details>

Done.
___
*[versioned/spi/src/main/java/com/dremio/nessie/versioned/ValueWorker.java, line 33 at r2](https://reviewable.io/reviews/projectnessie/nessie/701#-MRUnon-0dBfcRIffu2-:-MRWomwC3rDSbmrKl5XK:b-lpo32t) ([raw file](https://github.com/projectnessie/nessie/blob/828a32ac526dd09c07dba8fb8190546976b463c1/versioned/spi/src/main/java/com/dremio/nessie/versioned/ValueWorker.java#L33)):*
<details><summary><i>Previously, rymurr (Ryan Murray) wrote…</i></summary><blockquote>

> ```
> track multiple types of assets under a single AssetKey heading. 
> ```

I am not sure what you mean here. In my mind an iceberg table would return a stream of `AssetKey` with 1 key per individual file. When you say 'AssetKey heading' do you mean 1 asset key would have a set of related files?
</blockquote></details>

updated javadoc to better clarify.
___
*[versioned/tiered/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoBaseValue.java, line 96 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROvmB-1ax-2S4-p-3W:-MRWpGPcAwjqxWG_688d:b7uohsx) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoBaseValue.java#L96)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

Can you change this to either let all users of `writeInt64` use this method or specialize it to make clear it's only used for "timestamp in microseconds since epoc"?
</blockquote></details>

Updated to be used elsewhere.
___
*[versioned/tiered/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoL1.java, line 92 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROvBtSE7pixwMMUf9l:-MRWp_pl2DZA4tjqqx1J:bb7f2mf) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoL1.java#L92)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

Mind changing this to the `serializeLong()` function you've added?
</blockquote></details>

done
___
*[versioned/tiered/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoRef.java, line 236 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROvQX14vrPhX4e_-7j:-MRWpd2wDLVHNSgWDX9s:b-896fix) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoRef.java#L236)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

Mind changing this to the `serializeLong()` function you've added?
</blockquote></details>

Done.
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/Ids.java, line 20 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROpzKx0WxXDB7Jil06:-MRWpeV4-dHFEWcwmJfu:b-auj3tp) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/Ids.java#L20)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

The methods in this class are never used. Missing a refactoring?
</blockquote></details>

I added it for future use within GC.
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/Ids.java, line 20 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MRP2KnC1bbHN36YXaJk:-MRWpgVX2WdOqfKuYf1a:b-4zzt6u) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/Ids.java#L20)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

Since `DT`and `Ids` are rather small, internal classes that provide simple utility methods, mind combining those?
</blockquote></details>

Since DT is package private and Ids is public, I'm inclined to keep them separate. They also serve pretty different purposes.
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/InternalL1.java, line 197 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROsU5qB1moXHi4-l2q:-MRWprP32om8gIvpWN_U:b-896fix) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/InternalL1.java#L197)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

Unnecessarily added empty line
</blockquote></details>

Done.
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/InternalL1.java, line 221 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROytlN8cCFzRPLGN2e:-MRWqtem0O-zy7YD4dDL:b-896fix) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/InternalL1.java#L221)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

I think, both `id()` and `dt()`should be moved to `PersistentBase`.
I've pushed that here: https://github.com/snazy/nessie/commit/40113cce7325b95cece4fd9c8c064d5f0fc5f047
</blockquote></details>

Done.
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java, line 28 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROyiQM5Y6ycsi3T7FU:-MRWr11w6m0hjPHBGoIS:b69xcg7) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java#L28)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

Nit: unnecessary empty line
</blockquote></details>

i kept as one is now final and the other is not. Also clarifies that the comment only applies to id.
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java, line 61 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROr0Ka6jCZpjo7tqtw:-MRWrWmfADfjMpPERXlq:b-896fix) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java#L61)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

Implementations of this method are `public`, but should be package-private
</blockquote></details>

Done.
___
*[versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java, line 63 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROrCH62mu-QP15F0M9:-MRWrYkp8x8M0YZ0Ct2H:blzfvnj) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/PersistentBase.java#L63)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

This method isn't used.
</blockquote></details>

It is in the GC patch :)

Also, I think we should refactor other save op uses to use it in the main internal module.
___
*[versioned/tests/src/main/java/com/dremio/nessie/versioned/StringSerializer.java, line 60 at r1](https://reviewable.io/reviews/projectnessie/nessie/701#-MROwHJ5AjB0gQEfWaGN:-MRWnpl0482285BtQgK1:b-o46rus) ([raw file](https://github.com/projectnessie/nessie/blob/0849dec9a0e7a75a77434cd0dadbde4dfd4eedc1/versioned/tests/src/main/java/com/dremio/nessie/versioned/StringSerializer.java#L60)):*
<details><summary><i>Previously, snazy (Robert Stupp) wrote…</i></summary><blockquote>

The suppression is unnecessary
</blockquote></details>

already removed :)

**Review by rymurr — APPROVED** [review]
lgtm!

**Comment by snazy** [comment]
LGTM

**Review by snazy — APPROVED** [review]

**Pull request merged**